web/packages/teleport/src/Bots/Edit/EditDialog.tsx

```diff
@@ -0,0 +1,298 @@
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { useState } from 'react';
+
+import {
+  Alert,
+  Box,
+  ButtonPrimary,
+  ButtonSecondary,
+  Flex,
+  Indicator,
+} from 'design';
+import Dialog, {
+  DialogContent,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+} from 'design/DialogConfirmation';
+import FieldInput from 'shared/components/FieldInput';
+import { FieldSelectAsync } from 'shared/components/FieldSelect';
+import { Option } from 'shared/components/Select';
+import {
+  TraitsEditor,
+  TraitsOption,
+} from 'shared/components/TraitsEditor/TraitsEditor';
+import Validation from 'shared/components/Validation';
+import { requiredField } from 'shared/components/Validation/rules';
+
+import { editBot, fetchRoles } from 'teleport/services/bot/bot';
+import { EditBotRequest, FlatBot } from 'teleport/services/bot/types';
+import useTeleport from 'teleport/useTeleport';
+
+import { formatDuration } from '../formatDuration';
+import { createGetBotQueryKey, useGetBot } from '../hooks';
+
+export function EditDialog(props: {
+  botName: string;
+  onCancel: () => void;
+  onSuccess: (bot: FlatBot) => void;
+}) {
+  const { botName, onCancel, onSuccess } = props;
+  const queryClient = useQueryClient();
+  const ctx = useTeleport();
+  const flags = ctx.getFeatureFlags();
+  const hasReadPermission = flags.readBots;
+  const hasEditPermission = flags.editBots;
+
+  const [selectedRoles, setSelectedRoles] = useState<string[] | null>(null);
+  const [selectedTraits, setSelectedTraits] = useState<TraitsOption[] | null>(
+    null
+  );
+  const [selectedMaxSessionDuration, setSelectedMaxSessionDuration] = useState<
+    string | null
+  >(null);
+  const { isSuccess, data, error, isLoading } = useGetBot(
+    { name: botName },
+    {
+      enabled: hasReadPermission,
+      staleTime: 30_000, // Keep data in the cache for 30 seconds
+    }
+  );
+
+  const {
+    mutate,
+    error: saveError,
+    isPending: isSubmitting,
+  } = useMutation({
+    mutationFn: (params: EditBotRequest) => {
+      return editBot(ctx.getFeatureFlags(), botName, params);
+    },
+    onSuccess: newData => {
+      const key = createGetBotQueryKey({ name: newData.name });
+      queryClient.setQueryData(key, newData);
+
+      setSelectedRoles(null);
+      setSelectedTraits(null);
+      setSelectedMaxSessionDuration(null);
+
+      onSuccess(newData);
+    },
+  });
+
+  const handleSubmit = () => {
+    const roles = selectedRoles ?? null;
+    const traits =
+      selectedTraits?.map(t => ({
+        name: t.traitKey.value,
+        values: t.traitValues.map(v => v.value),
+      })) ?? null;
+    const max_session_ttl =
+      selectedMaxSessionDuration?.trim()?.replaceAll(' ', '') ?? null;
```

Comment:
```suggestion
      selectedMaxSessionDuration?.trim().replaceAll(' ', '') ?? null;
```
The result returned from `trim()` is always a string.